internal/handlers/common/filter.go

```diff
@@ -1084,3 +1093,35 @@ func filterFieldValueByTypeCode(fieldValue any, code typeCode) (bool, error) {
 
 	return true, nil
 }
+
+// filterFieldExprElemMatch handles {field: {$elemMatch: value}}.
+// Returns false if doc value is not an array.
+// TODO: https://github.com/FerretDB/FerretDB/issues/364
+func filterFieldExprElemMatch(doc *types.Document, filterKey string, exprValue any) (bool, error) {
+	value := must.NotFail(doc.Get(filterKey))
+
+	if _, ok := value.(*types.Array); !ok {
+		return false, nil
+	}
+
+	expr, ok := exprValue.(*types.Document)
+	if !ok {
+		return false, NewErrorMsg(ErrBadValue, "$elemMatch needs an Object")
+	}
+
+	for _, key := range expr.Keys() {
+		if slices.Contains([]string{"$text", "$where"}, key) {
+			return false, NewErrorMsg(ErrBadValue, fmt.Sprintf("%s can only be applied to the top-level document", key))
+		}
+
```

Comment:
Would you please add a TODO with the issue number here? Thank you.

Reply:
Added